crates/common/src/oracle/vec.rs

```diff
@@ -197,12 +411,363 @@ impl PreimageOracleClient for VecOracle {
 
 #[async_trait]
 impl HintWriterClient for VecOracle {
+    /// Asynchronously writes data or performs an operation based on the provided hint.
+    ///
+    /// This function serves as a placeholder implementation that currently does nothing
+    /// and always returns `Ok(())`.
+    ///
+    /// # Notes
+    /// This function is currently a no-op and may be extended in the future to perform
+    /// meaningful write operations based on the provided hint.
     async fn write(&self, _hint: &str) -> PreimageOracleResult<()> {
         Ok(())
     }
 }
 
+/// Reads and deserializes a shard into a `PreimageVecEntry` structure.
+///
+/// This function retrieves binary data representing a serialized shard from the environment.
+/// It then attempts to deserialize the binary data into a `PreimageVecEntry` instance using the
+/// `rkyv` deserialization framework. If the deserialization process fails, the function will panic
+/// with an error message.
+///
+/// # Returns
+/// - A `PreimageVecEntry` object that represents the deserialized shard.
+///
+/// # Panics
+/// - The function panics if deserialization fails, with the message `"Failed to deserialize shard"`.
+///
+/// # Dependencies
+/// This function uses:
+/// - `env::read_frame()` to read binary data from the environment.
+/// - `rkyv::from_bytes` for deserialization of the binary data into a `PreimageVecEntry`.
+///
+/// Ensure that the environment contains valid binary data for a `PreimageVecEntry` structure before
+/// calling this function.
+#[cfg(target_os = "zkvm")]
```

Comment:
Hmmm, this uses riscv0_zkvm::guest::env inside a core library. Perhaps consider adding a helper function that reads the entry from a frame, and performing the frame reading in the guest itself.